mindsdb/integrations/handlers/zendesk_handler/zendesk_tables.py

```diff
@@ -0,0 +1,487 @@
+import pandas as pd
+from typing import List
+from mindsdb.integrations.libs.api_handler import APITable
+from mindsdb.integrations.utilities.handlers.query_utilities import SELECTQueryParser, SELECTQueryExecutor
+from mindsdb.utilities import log
+from mindsdb_sql.parser import ast
+
+logger = log.getLogger(__name__)
+
+
+class ZendeskUsersTable(APITable):
+    """Zendesk Users Table implementation"""
+
+    def select(self, query: ast.Select) -> pd.DataFrame:
+        """Pulls data from the zendesk list users API
+
+        Parameters
+        ----------
+        query : ast.Select
+           Given SQL SELECT query
+
+        Returns
+        -------
+        pd.DataFrame
+            Zendesk users
+
+        Raises
+        ------
+        ValueError
+            If the query contains an unsupported condition
+        """
+
+        select_statement_parser = SELECTQueryParser(
+            query,
+            'users',
+            self.get_columns()
+        )
+
+        selected_columns, where_conditions, order_by_conditions, result_limit = select_statement_parser.parse_query()
+
+        subset_where_conditions = []
+        for op, arg1, arg2 in where_conditions:
+            if arg1 in self.get_columns():
+                subset_where_conditions.append([op, arg1, arg2])
+
+        zen_users = list(self.handler.zen_client.users())
```

Comment:
Is it at all possible to avoid querying all of the tickets and apply the conditions + the limit clause here? This would improve performance? It would be great if the same can be done for the other resources as well.